src/script/lua_api/l_nodemeta_local.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "lua_api/l_nodemeta_local.h"
+#include "inventory.h"
+#include "lua_api/l_internal.h"
+#include "lua_api/l_inventory.h"
+#include "common/c_content.h"
+//#include "lua_api/l_inventory.h"
+/*
+	NodeMetaRef
+*/
+NodeMetaLocalRef* NodeMetaLocalRef::checkobject(lua_State *L, int narg)
+{
+	luaL_checktype(L, narg, LUA_TUSERDATA);
+	void *ud = luaL_checkudata(L, narg, className);
+	if(!ud) luaL_typerror(L, narg, className);
+	return *(NodeMetaLocalRef**)ud;  // unbox pointer
+}
+
+Metadata* NodeMetaLocalRef::getmeta(bool auto_create)
+{
+	return m_meta;
+}
+
+void NodeMetaLocalRef::clearMeta()
+{
```

Comment:
i'm not sure clang-format like it, it's not very important. Also if it's really empty just keep it in header file